### DISCUSSION TITLE
Section 3 : Unable to find the .pandoc file on my route directory, after downloading pandoc ?

### DISCUSSION THREAD

**Discussion by AnouarBF** [Cyfrin/security-and-auditing-full-course-s23#31] [Q&A]
I downloaded pandoc and run the command `pandoc --help` to check if it works, and indeed it does work. but when I try to get into .pandoc file from my computer it shows "No such file or directory".
By the way I tried this method https://github.com/Cyfrin/security-and-auditing-full-course-s23/discussions/15#discussion-5941217 from @kwamentw but it doesn't work.
I'm using Ubuntu 22.04

**Comment by Ayodeji63** [comment]
i ran into same issue, what i did was i created the directory myself then add the eisvogel.latex file

**Reply by kwamentw** [reply]
@AnouarBF  try this solution too & let's see. If it doesn't work then you send a markdown of your command and the error, so we can help you out

**Reply by AnouarBF** [reply]
@Ayodeji63  did it work with you ??

**Reply by kwamentw** [reply]
He said his solution worked

**Reply by AnouarBF** [reply]
@kwamentw No it didn't work as well. it shows me this error `Could not find data file /usr/share/pandoc/data/templates/eivogel.latex` after runnint this command :
```
pandoc report.md -o report.pdf --from markdown --template=eivogel --listings
```
maybe its because of the way I installed pandoc and latex. I installed them like that:
```
sudo apt install pandoc
```
```
sudo apt install texlive
```
Is there another way to install them on Ubuntu 22.04 ?

**Reply by Ayodeji63** [reply]
Yes, the problem is with the way you installed `pandoc` and `texlive` i also did that and it didn't work. 

So follow this installation pattern

## Pandoc

1. Purge apt installed, you must "undo" the bad instruction.
```bash
sudo apt purge pandoc
```
2. Get fresh .deb at [Pandoc's git/releases](https://github.com/jgm/pandoc/releases):
```bash 
wget https://github.com/jgm/pandoc/releases/download/3.1.9/pandoc-3.1.9-1-amd64.deb
```
3. Then Install:
 ```bash 
sudo dpkg -i pandoc-3.1.9-1-amd64.deb
```
   
  ## Texlive
  This will require lots of internet-data i used about 3GB to install it because, we are installing the full file, it maybe less or more at your end
 
  1. Install TeX Live
```bash 
sudo apt-get install texlive-full
```
    
  2. Install Texmaker
```bash
sudo apt-get install texmaker
```
      

After the try running this again
```bash 
pandoc report.md -o report.pdf --from markdown --template=eivogel --listings
```

Let me know if you ran into any issue.

**Reply by AllenGeorge08** [reply]
Hey @Ayodeji63 I tried your approach but still couldn't find the .pandoc.. I downloaded the latest amd version in the pandoc repo.

**Reply by dharminnagar** [reply]
@Ayodeji63 I did this too, worked perfectly for me when I was starting out :)

**Comment by kwamentw** [comment]
@AnouarBF, Firstly your spelling of the markdown file is wrong, it is `eisvogel.latex` not `eivogel.latex`.
Secondly, make sure you have downloaded the `eisvogel.latex` file from  patricks auditing template repo `https://github.com/Cyfrin/audit-report-templating/blob/main/eisvogel.latex` 
When done try any of the two solutions again, either you pass the path of the downloaded file like i did or you create the .pandoc folder and move the eisvogel file into it.

try this and lets see

**Comment by DuncanDuMond** [comment]
@kwamentw I tried following your example and it does not work. I could not find the ./pandoc folder, then used the command you recommended and got this prompt:
```
Could not find data file templates/Solidity.latex
```

Could not find data file templates/Solidity.latex
I will also attach how the directory looks from my Mac:
<img width="337" alt="Screenshot 2023-12-20 at 12 56 20 PM" src="https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/71061346/37c4797b-88ad-4cda-887a-af9563e9f83e">

Please advise

**Reply by Ayodeji63** [reply]
Which code did you run?

**Reply by DuncanDuMond** [reply]
I ran:
```
pandoc report.md -o report.pdf --from markdown --template=eisvogel --listings
```

and 
```
pandoc report.md -o report.pdf --from markdown --template=~/Downloads/eisvogel --listings  
```

**Reply by dharminnagar** [reply]
The `.pandoc` folder should be in the root directory not in the project directory

**Comment by kwamentw** [comment]
Okay, you have to first find the path of your `eisvogel.latex` and insert it into the template flag like I did `template=~/Downloads/eisvogel`
The reason I have that path is I downloaded the file to my downloads directory. So for yours to work:

1. Make sure you have downloaded the file
2. Find the path of your file 
3. Pass the path into the template.
4. Re run the command 

Let's see whether this helps,
Will be expecting your feedback.

**Reply by hamzalagrari** [reply]
hello i have the same problem i can find the .pandoc file , should i create .pandoc file in my home directory ot i should created inside audi-data folder ? thx for your time bro

**Reply by kwamentw** [reply]
Read @Ayodeji63 solution above`https://github.com/Cyfrin/security-and-auditing-full-course-s23/discussions/31#discussioncomment-7894257`. I think it's his solution you're following.

If you want another alternative follow my solution `https://github.com/Cyfrin/security-and-auditing-full-course-s23/discussions/15#discussion-5941217`

**Reply by hamzalagrari** [reply]
i did follow his solution but it didnt work , i just want to know where did you put the .pandoc file in home directory or in audit-data folder ?

**Reply by Ayodeji63** [reply]
The home directory

How did you install pandoc and latex?

**Reply by hamzalagrari** [reply]
i install it with the same cmd that you mentioned "https://github.com/Cyfrin/security-and-auditing-full-course-s23/discussions/31#discussioncomment-7894257" but i get this error : 'could not find data file templates/eisvogel.latex'

**Reply by kwamentw** [reply]
have you created the .pandoc in the home directory as stated?

**Reply by hamzalagrari** [reply]
yes ,then should i create templates inside the .pandoc folder because i created .pandoc manually with 'mkdir .pandoc' , because i coudnt find .pandoc ? i hope you understand my problem

**Reply by DuncanDuMond** [reply]
I am having the same problem myself

**Comment by DuncanDuMond** [comment]
> Okay, you have to first find the path of your `eisvogel.latex` and insert it into the template flag like I did `template=~/Downloads/eisvogel` The reason I have that path is I downloaded the file to my downloads directory. So for yours to work:
> 
> 1. Make sure you have downloaded the file
> 2. Find the path of your file
> 3. Pass the path into the template.
> 4. Re run the command
> 
> Let's see whether this helps, Will be expecting your feedback.

I tried your recommendation and it still does not work.

I still get this prompt:
```
Could not find data file templates/eisvogel.latex
```

**Reply by kwamentw** [reply]
Okay, please send the location of your eisvogel.latex file, the command you run & the error that comes after running the command.

Let me see whether I can help

**Reply by DuncanDuMond** [reply]
The path is as follows:

Downloads/Solidity_Practice/security-course/3-passwordstore-audit/audit-data/.pandoc/templates/eisvogel.latex

See above jpeg

**Reply by kwamentw** [reply]
try the following command & make sure you're in the right directory that is the audit-data directory

`pandoc report.md -o report1.pdf  --from markdown  --template=~/Downloads/Solidity_Practice/security-course//3-passwordstore-audit/audit-data//.pandoc/templates/eisvogel.latex  --listings`

if this doesn't work do well to send the error message 

but why do you have double '//' in your path?

**Reply by kwamentw** [reply]
@DuncanDuMond have you tried this ?

**Reply by DuncanDuMond** [reply]
Still no luck.

Here is the error message:
```
Could not find data file templates/eisvogel.latex
```

**Reply by kwamentw** [reply]
why did you have double slashes `//` in the paths? 

Is it a Mac thing? If it's not please make it single/normal and try the command again let's see

**Reply by DuncanDuMond** [reply]
No it was an error in my part. It was supposed to be single slashes....trying again

**Reply by DuncanDuMond** [reply]
Still get the same error after fixing the prompt....SMH

**Reply by DuncanDuMond** [reply]
Could it be because I am trying to do the prompt from the audit folder project folder instead of a different folder?

**Reply by kwamentw** [reply]
Try put eisvogel in the downloads folder and try again let's see

**Reply by DuncanDuMond** [reply]
I have tried that, made a copy in the Downloads/templates folder, and in a Downloads/.pandoc/templates...still nothing

**Reply by kwamentw** [reply]
Try just the Downloads folder, don't create any folder in the downloads folder. Keep the path as simple as possible

If it doesn't work, uninstall pandoc and TexLive

Then follow the link Patrick put in the repo to reinstall pandoc & Texlive

Then try again let's see

**Reply by DuncanDuMond** [reply]
Still no luck after uninstalling and reinstalling unfortunately.

**Reply by kwamentw** [reply]
Did you use the links provided in Patrick's repo to install?

**Reply by DuncanDuMond** [reply]
Yes I did

**Reply by kwamentw** [reply]
Still no luck? Then you have to relax and probe your error by yourself, use AI, google and other platforms. You'll find a solution by digging and understanding the situation yourself.

**Reply by AnouarBF** [reply]
@DuncanDuMond I'm also witnessing the same problem you have, if you have found any help please share it. as well as me. This thing is not stopping us buddy !
BTW is there any other template we can use as alternative ??

**Reply by DuncanDuMond** [reply]
Not sure yet as I am currently on Section 4. My guess is there are other formats as you get further along the course. My only thing I can think of that is causing the issue is due to pandoc and/or LateX not being compatible with macOS Sonoma. I could be wrong but that was the issue with installing homebrew for a while.

**Reply by kwamentw** [reply]
If it's an issue with the OS try using VMware or any virtual machine setup with Linux

**Reply by kwamentw** [reply]
@AnouarBF @DuncanDuMond check this solution out whether it will work
https://github.com/Cyfrin/security-and-auditing-full-course-s23/discussions/15#discussioncomment-7988247

**Comment by EngrPips** [comment]
@AnouarBF @DuncanDuMond  do you guys get the pandoc problem resolved already ?

**Reply by Naties29** [reply]
Hi Engr, I am having issues with this as well. Attempting one more fix and seeing if the problem is resolved. 

Both Pandoc and the alternate option the Markdown PDF VSC extension both are failing for me. Ill update this thread with more info once I attempt this final fix.

**Reply by EngrPips** [reply]
Alright, Looking forward to seeing your outcome.

**Reply by Naties29** [reply]
Unfortunately, I haven't had any luck. I continually receive this error when trying to generate the report. I have tried the above fixes but nothing is working as of now. 

I could not find data file templates/Vogel.latex

Curiously when I navigate to my home folder, there is no pandoc folder being displayed like Patrick states in his video, but this is shown when I type in pandoc --version

pandoc 3.1.9
Features: +server +lua
Scripting engine: Lua 5.4
User data directory: /home/naties/.local/share/pandoc
Copyright (C) 2006-2023 John MacFarlane. Web: https://pandoc.org
This is free software please take a look at the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.

This is what I receive whenavigatingte to that file pats stated in the pandoc help output.

bash: cd: /home/naties/.local/share/pandoc: No such file or directory

Thank you for your time Mr Pips!

**Reply by EngrPips** [reply]
I faced the same issue, too, so I deleted the pandoc I had installed and followed the guide on this site, https://pandoc.org/installing.html, to install pandoc. Everything just worked out of the box. How did you install pandoc initially?

**Reply by Naties29** [reply]
I have installed it the exact way from the website and still no luck, will continue to keep trying. Not sure what I am missing here.

I move the DEB into my home folder, install the deb like the pandoc website says and still nothing. Will continue to troubleshoot.

**Reply by EngrPips** [reply]
Can you maybe ask AI buddy and see if they will have something meaningful that might help

**Reply by FrankezeCode** [reply]
> I faced the same issue, too, so I deleted the pandoc I had installed and followed the guide on this site, https://pandoc.org/installing.html, to install pandoc. Everything just worked out of the box. How did you install pandoc initially?

@EngrPips  bro, How did you follow the instruction in this link, Is confusing to me. 
And which operating system are  you installing on, bcos I'm using wsl and the option is not there

**Reply by EngrPips** [reply]
I am operating on Mac OS, and it was kind of a straight instruction for Mac

**Comment by ProKing13** [comment]
Bro i have solved the problem like below:

@EngrPips @kwamentw @AnouarBF 
first i have copied  the eisvogel.tex file and paste it in this directory: 
```
 /usr/share/pandoc/data/templates/
```
and then i have installed some package through the below command:
```
sudo apt install texlive-full
```
then manually i give the path of the file and run the below command and it worked for me:
```
pandoc report-example.md -o report.pdf --from markdown --template=/usr/share/pandoc/data/templates/eisvogel.tex --listings
```

**Reply by EngrPips** [reply]
Awesome, I am glad you figured this out. It would be useful for anyone who runs into the same problem in the future.

**Comment by Keivou** [comment]
Hey guys, one thing you can do is running the base command with `-vvvv` at the end:

```bash
pandoc report.md -o report.pdf --from markdown --template=eisvogel --listings -vvvv
```

This will return where the system is looking for the `templates` folder, it should print something like this:

```
pandoc.exe 3.6.1
User data directory: C:\<User>\AppData\Roaming\pandoc
Copyright (C) 2006-2024 John MacFarlane. Web:  https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
```

You can then navigate to that `pandoc` folder, create a `templates` folder, add the `eisvogel.latex` file and you should be good to go!

**Reply by EngrPips** [reply]
Thanks for the heads-up @PedroSolorzano
